### DISCUSSION TITLE
Can I embed inline code?

### DISCUSSION THREAD

**Discussion by jpvelez** [quarto-dev/quarto-cli#8681] [Q&A]
### Description

Hi there,

Just stated using 1.4, and I think manuscripts and embeds are totally brilliant. Being able to separate computation into a companion notebook and writing into `index.md` is an awesome pattern.

I love the ability to embed a cross-ref (figures, table) from a companion notebooks using `{{{< render companion.md#fig-name >}}}`.

Is it possible to do the same for inline code? Before 1.4, we would generate our the datasets and variables to be used in our article in a separate `.qmd` file, and then export them to disk. In order to make our article update to reflect our upstream analysis, we would import the datasets and write chart code to generate charts, and import variables for any numbers that we needed to inject dynamically into the body of the text, and do so using inline code syntax.

With 1.4, we can just generate the charts directly in the companion notebook and simply embed them. But is there a way to do the same for individual variables in the companion notebook that you want to embed into the middle of a sentence in the `index.md` notebook? I tried a few different things and couldn't figure it out.

If this feature doesn't exist yet, it strikes me as essential for achieving full separation between companion and index notebooks in manuscript projects. We use inline code all the time. (Embedded inlines might need a different design treatment, so that the "Source: <companion notebook>" bit wouldn't break up the flow of text.)

**Comment by mcanouil** [comment]
Would you mind sharing a Git repository with more concrete elements.
You are using some vocabulary that has no meaning in Quarto.
I am also not completely sure what you mean by "inline code". Is it `` `{python} print("Hello, World!")` ``?

Where did you see `{{{< render companion.md#fig-name >}}}`? I am pretty sure, this does not exist, but `{{< embed ... >}}` does (https://quarto.org/docs/authoring/notebook-embed.html).

**Comment by jpvelez** [comment]
- Yes, it should be `render` not `embed`.
- Yes, that's what I mean by inline code.
- Sharing a git repo wouldn't really help, because I'm describing something that currently isn't possible. To rephrase: it's possible to embed figures and tables from companion numbers. But what if you need to embed a number into the middle of a sentence, the way you might do with `{python} print("Hello, World!")`? Does that make sense?

**Reply by cscheid** [reply]
That feature doesn't exist yet.

**Reply by jpvelez** [reply]
Got it!

Is this something ya'll are planning? Would be excited to use it!

**Comment by edenian-prince** [comment]
I agree with @jpvelez it would be a great feature to have. @mine-cetinkaya-rundel discusses this feature in her tutorials but after digging it looks like the inline code only comes from the main index.qmd file. See 23:50 https://www.youtube.com/watch?v=NK1onTLcgY4&ab_channel=RConsortium and the repo here https://github.com/mine-cetinkaya-rundel/indo-rct/blob/0d43b62ea5d30b12a9cfc82e0eeb52b16840e34c/index.qmd#L193C1-L204C32 

I think there are some workarounds. If I figure it out I'll keep you updated
